tests/test_hs015.py

```diff
@@ -193,6 +195,90 @@ def test_snopt_snstop(self):
         # we should get 70/74
         self.assert_inform_equal(sol, optInform=74)
 
+    def test_snopt_snstop_restart(self):
+        pickleFile = "restart.pickle"
+
+        def my_snstop_restart(iterDict, restartDict):
+            # Save the restart dictionary
+            writePickle(pickleFile, restartDict)
+
+            # Exit after 5 major iterations
+            if iterDict["nMajor"] == 5:
+                return 1
+
+            return 0
+
+        # Run the optimization for 5 major iterations
+        self.optName = "SNOPT"
+        self.setup_optProb()
+        optOptions = {
+            "snSTOP function handle": my_snstop_restart,
+            "snSTOP arguments": ["restartDict"],
+        }
+        sol = self.optimize(optOptions=optOptions, storeHistory=True)
+
+        # Check that the optimization exited with 74
+        self.assert_inform_equal(sol, optInform=74)
+
+        # Read the restart dictionary pickle file saved by snstop
+        restartDict = readPickle(pickleFile)
+
+        # Now optimize again but using the restart dictionary
+        self.setup_optProb()
+        opt = OPT(
```

Comment:
You are not using `self.optimize()` just to avoid all the extra checks or am I missing some initialization step?

Reply:
`self.optimize()` does not have a way of passing in `restartDict`